lib/parsers/heif_parser.rb

```diff
@@ -0,0 +1,430 @@
+    safe_read(@buf, 2) # we skip the first 2 bytes of the flags (total of 3 bytes) cause we care only about the least significant bit
+    flags = read_int_8
+    entry_count = read_int_32
+    item_id = 0
+    entry_count.times do
+      item_id = if version == 0
+        read_int_16
+      else
+        read_int_32
+      end
+
+      association_count = read_int_8
+      association_count.times do
+        # we need to retrieve the "essential" bit wich is just the first bit in the next byte
+        binary = convert_byte_to_binary(read_int_8)
+        # essential_bit = binary[0] # uncomment if needed
+        binary.concat(convert_byte_to_binary(read_int_8)) if (flags & 1) == 1 # if flag is 1 we need the next 15 bits instead of only the next 7 bits
+        # we need to nullify the 1st bit since that one was the essential bit and doesn't count now to calculate the property index
+        binary[0] = 0
+        item_property_index = binary.join.to_i(2)
+        # we are interested only in the primary item properties
+        @item_props_idxs << item_property_index if item_id == @primary_item_id
+      end
+
+      # we are interested only in the primary item
+      if item_id != @primary_item_id
+        next
+      else
+        return
+      end
+    end
+  end
+
+  def fill_primary_values
+    @item_props_idxs.each { |x|
+      next if @item_props[x].nil?
+      prop = @item_props[x]
+      case prop[:type]
+      when IMAGE_SPATIAL_EXTENTS_BOX
+        @width = prop[:width]
+        @height = prop[:height]
+      when PIXEL_ASPECT_RATIO_BOX
+        @pixel_aspect_ratio = prop[:pixel_aspect_ratio]
+      when COLOUR_INFO_BOX
+        @colour_info = prop[:colour_info]
+      when PIXEL_INFO_BOX
+        @pixel_info = prop[:pixel_info]
+      when RELATIVE_LOCATION_BOX
+        @horizontal_offset = prop[:horizontal_offset]
+        @vertical_offset = prop[:vertical_offset]
+      when CLEAN_APERTURE_BOX
+        @clean_aperture = prop[:clean_aperture]
+      when IMAGE_ROTATION_BOX
+        @rotation = prop[:rotation]
+      end
+    }
+  end
+
+  def next_meaningful_meta_byte
+    while @buf.pos < @metadata_end_pos
+      next_byte = read_string(4)
+      return next_byte if meaningful?(next_byte)
+    end
+  end
+
+  def get_next_box(box_start_pos, box_length, end_pos_upper_box)
+    skip_pos = box_start_pos + box_length - HEADER_LENGTH
+    @buf.seek(skip_pos)
+    return if skip_pos >= end_pos_upper_box
+    next_box_length = read_int_32
+    next_box_name = read_string(4)
+    [next_box_length, next_box_name, @buf.pos]
+  end
+
+  def meaningful?(byte)
+    byte != MEANINGLESS_BYTE
+  end
+
+  def convert_byte_to_binary(integer)
+    binary = []
+    while integer > 0
+      binary << integer % 2
+      integer /= 2
+    end
+    binary_value = binary.reverse
+    (8 - binary_value.length).times do
+      binary_value.prepend('0')
+    end
+    binary_value
+  end
+
+  def likely_match?(filename)
+    filename =~ /\.(heif|heic)$/i
```

Comment:
You can shorten this a bit:

```suggestion
    filename =~ /\.hei[cf]$/i
```